tests/unit/records/test_schemas.py

```diff
@@ -72,3 +72,37 @@ def test_contributors(app, db, minimal_record):
         {'name': 'test', 'affiliation': 'test', 'type': 'Invalid'}
     ]
     pytest.raises(ValidationError, Record.create, minimal_record)
+# validation of full_record fixture
+
+
+def test_full_json(app, db, full_record):
+    """Test full json."""
+    Record.create(full_record)
+
+
+def full_test_recid(app, db, full_record):
+    """Test recid property."""
+    full_record['recid'] = '12345'
+    pytest.raises(ValidationError, Record.create, full_record)
+
+
+def full_test_resource_type(app, db, full_record):
+    """Test resource type."""
+    full_record['resource_type'] = 'publication'
+    pytest.raises(ValidationError, Record.create, full_record)
+    full_record['resource_type'] = {'type': 'publication', 'subtype': 'book'}
+    Record.create(full_record)
+
+
+def full_test_contributors(app, db, full_record):
+    """Test contributors."""
+    full_record['contributors'] = [
+       {'affiliation': 'CERN', 'name': 'Smith, Other', 'type': 'Other',
+                       'gnd': '', 'orcid': '0000-0002-1825-0097'}
+    ]
+    Record.create(full_record)
+    full_record['contributors'] = [
+        {'affiliation': '', 'name': 'Hansen, Viggo', 'type': 'Other',
+                            'gnd': '', 'orcid': ''}
+    ]
+    pytest.raises(ValidationError, Record.create, full_record)
```

Comment:
This will not fail because the input is valid. See the specification of the "contributors": https://github.com/zenodo/zenodo/blob/master/zenodo/modules/records/jsonschemas/records/record-v1.0.0.json

Try "type" that's not in an enum. Then below in the same test try skipping the "name" field (with a proper "enum").